assignments/tasks.go

```
@@ -0,0 +1,284 @@
+		Title:        task.Title,
+		Body:         task.Body,
+	}
+	return issue, nil
+}
+
+// Following is Oje code (placement might be temporary):
+
+func getTasksFromAssignments(c context.Context, assignments []*pb.Assignment) []*pb.Task {
+	tasks := []*pb.Task{}
+	for _, assignment := range assignments {
+		tasks = append(tasks, assignment.Tasks...)
+	}
+
+	return tasks
+}
+
+func handleTasks(c context.Context, db database.Database, s scm.SCM, course *pb.Course, tasks []*pb.Task) error {
+	repos, err := db.GetRepositoriesWithIssues(&pb.Repository{
+		OrganizationID: course.GetOrganizationID(),
+	})
+	if err != nil {
+		return err
+	}
+
+	assignments, err := db.GetAssignmentsByCourse(course.GetID(), false)
+	if err != nil {
+		return err
+	}
+
+	// Loops through all assignments
+	for _, assignment := range assignments {
+		err := synchronizeTasks(c, db, assignment, tasks)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Loops through all student repos
+	for _, repo := range repos {
+		if !repo.IsStudentRepo() {
+			continue
+		}
+
+		err = synchronizeIssues(c, db, course, s, repo, tasks)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// synchronizeTasks synchronizes tasks in the database, with the ones of given assignment.
+// Returns a slice of tasks as they appear in the database.
+func synchronizeTasks(c context.Context, db database.Database, assignment *pb.Assignment, tasks []*pb.Task) error {
+	tasksToBeCreated := []*pb.Task{}
+	tasksToBeUpdated := []*pb.Task{}
+	taskMap := make(map[string]*pb.Task)
+	for _, task := range tasks {
+		if task.AssignmentOrder == assignment.Order {
+			taskMap[task.Name] = task
+		}
+	}
+
+	existingTasks, err := db.GetTasks(&pb.Task{AssignmentID: assignment.GetID()})
+	if err != nil {
+		return err
+	}
+
+	for _, existingTask := range existingTasks {
+		task, ok := taskMap[existingTask.Name]
+		if !ok {
+			// There exists a task in db, that is not represented by a task found in scm.
+			db.DeleteTask(existingTask)
```

Should use `_ = db.DeleteTask ...` to indicate that we ignore the error explicitly.